examples/07_join_aggregation.py

```diff
@@ -0,0 +1,387 @@
+# also consider aggregating by **movies**.
+#
+# Here, we compute the histogram of the target with 3 bins, before joining it
+# back on the initial table.
+from skrub import AggTarget
+
+
+agg_target = AggTarget(
+    main_keys="userId",
+    suffixes="_user",
+    operations=["hist(3)"],
+)
+X_transformed = agg_target.fit_transform(X, y)
+
+print(X_transformed.shape)
+print(X_transformed.head())
+
+###############################################################################
+# Chaining everything together in a pipeline
+# ------------------------------------------
+#
+# To perform cross-validation and enable hyper-parameter tuning, we gather
+# all elements into a scikit-learn |Pipeline| by using |make_pipeline|,
+# and define a scikit-learn |HGBR|.
+#
+# Since the auxiliary tables of |AggJoiner| are the main table itself, we need
+# to use the output of the previous layer (here the |TableVectorizer|).
+# We enable this behaviour by **using the placeholder** ``"X"`` **instead of a
+# dataframe** in the ``tables`` tuple.
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.pipeline import make_pipeline
+
+# Extracting datetime attributes from timestamps.
+table_vectorizer = TableVectorizer(
+    datetime_transformer=DatetimeEncoder(add_day_of_the_week=True)
+)
+table_vectorizer.set_output(transform="pandas")
+
+# Extracting features by aggregating datetime attributes.
+agg_joiner_user = AggJoiner(
+    tables=[
+        ("X", "userId", timestamp_cols),
+    ],
+    main_keys="userId",
+    suffixes="_user",
+    operations=["value_counts"],
+)
+
+agg_joiner_movie = AggJoiner(
+    tables=[
+        ("X", "movieId", timestamp_cols),
+    ],
+    main_keys="movieId",
+    suffixes="_movie",
+    operations=["value_counts"],
+)
+
+# Extracting features by aggregating the target
+agg_target_user = AggTarget(
+    main_keys="userId",
+    suffixes="_user",
+    operations=["value_counts"],
+)
+
+agg_target_movie = AggTarget(
+    main_keys="movieId",
+    suffixes="_movie",
+    operations=["value_counts"],
+)
+
+# Chaining all operations together
+pipeline = make_pipeline(
+    table_vectorizer,
+    agg_joiner_user,
+    agg_joiner_movie,
+    agg_target_user,
+    agg_target_movie,
+    HistGradientBoostingRegressor(learning_rate=0.1, max_depth=4, max_iter=40),
+)
+
+pipeline
+
+###############################################################################
+# Hyper-parameters tuning and cross validation
+# --------------------------------------------
+#
+# We can finally create our hyper-parameter search space, and use a
+# |GridSearchCV|. We select the cross validation splitter to be
+# the |TimeSeriesSplit| to prevent leakage, since our data are timestamped
+# logs.
+#
+# Note that you need the name of the pipeline elements to assign them
+# hyper-parameters search.
+#
+# You can lookup the name of the pipeline elements by doing:
+list(pipeline.named_steps)
```

Comment:
Because it returns a dict with a lot of elements. If I'm only interested in the name of the steps, I need to fetch the keys. I prefer `list(pipeline.named_steps)` over `pipeline.named_steps.keys()`. WDYT?